### PR TITLE
Add bulk results uploader support for genders of sexual partners column

### PR DESCRIPTION
**Pull request by mpbrown**
# BACKEND PULL REQUEST

## Related Issue

- Part of #7455

## Changes Proposed

- Adds support for an optional `genders_of_sexual_partners` column in the bulk results uploader

## Additional Information

- Following PR will add `patient_gender_identity` column
- Following PR will add logic for conditionally requiring `genders_of_sexual_partners` and `pregnant` columns if a row has a positive HIV test result
- Ticket for updating the user guidance is #7461 

## Testing

- Deployed on dev6

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java [hunk 1]
```diff
@@ -829,4 +829,25 @@ public static final Map<String, String> getResidenceTypeMap() {
           TRANS_WOMAN, "Female gender identity",
           OTHER, "Non-binary gender identity",
           REFUSED, "Asked But Declined");
+
+  public static Map<String, String> getGenderIdentityAbbreviationMap() {
```
Comment (fzhao99) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java:833]: nit: could keep the existing style for instantiating the maps here consistent and use `genderIdentityAbbrevationMap = Map.of(...)`?

**Review by fzhao99 — APPROVED** [review]
left one nit but lgtm!

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java (hunk 1 above)
Comment (mpbrown) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java:833]: IIRC using `Map.of(...)` in that way only works for up to 10 elements :(

Comment (fzhao99) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java:833]: Think we might be able to use `Map.ofEntries` in that case! 

Reading up on how that method is implemented under the hood made my jaw drop. 
Screenshot 2024-03-26 at 9 50 39 AM

**Review by bobbywells52 — APPROVED** [review]
LGTM -- great work on this!

**Pull request merged**